File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Types/StoreTopLevelPrimitiveTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.Tests.Common.Types;

namespace Db4objects.Db4o.Tests.Common.Types
{
	public class StoreTopLevelPrimitiveTestCase : AbstractDb4oTestCase
	{
		public static void Main(string[] args)
		{
			new StoreTopLevelPrimitiveTestCase().RunAll();
		}

		public virtual void Test()
		{
			bool exceptionHappened = false;
			try
			{
				Store(42);
			}
			catch (ObjectNotStorableException onsex)
			{
				exceptionHappened = true;
				StringAssert.Contains("Value types", onsex.Message);
			}
			Assert.IsTrue(exceptionHappened);
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/types/StoreTopLevelPrimitiveTestCase.java
/* Copyright (C) 2011 Versant Inc. http://www.db4o.com */
package com.db4o.db4ounit.common.types;

import com.db4o.ext.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class StoreTopLevelPrimitiveTestCase extends AbstractDb4oTestCase {
	
	public static void main(String[] args) {
		new StoreTopLevelPrimitiveTestCase().runAll();
	}
	
	public void test() {
		boolean exceptionHappened = false;
		try {
			store(new Integer(42));
		}
		catch(ObjectNotStorableException onsex){
			exceptionHappened = true;
			StringAssert.contains("Value types", onsex.getMessage());
		}
		
		Assert.isTrue(exceptionHappened);
	}
}
